temporal/api/cloud/cloudservice/v1/service.proto

```diff
@@ -125,4 +125,49 @@ service CloudService {
             get: "/api/v1/cloud/regions/{region}",
         };
     }
+
+    // Get all groups
+    rpc GetUserGroups (GetUserGroupsRequest) returns (GetUserGroupsResponse) {
+        option (google.api.http) = {
+            get: "/api/v1/cloud/user-groups",
+        };
+    }
+
+    // Get a group
+    rpc GetUserGroup (GetUserGroupRequest) returns (GetUserGroupResponse) {
+        option (google.api.http) = {
+            get: "/api/v1/cloud/user-groups/{group_id}",
+        };
+    }
+
+    // Create new a group
+    rpc CreateUserGroup (CreateUserGroupRequest) returns (CreateUserGroupResponse) {
+        option (google.api.http) = {
+            post: "/api/v1/cloud/user-groups",
```

Comment:
Comments from @cretz
>It is a bit confusing that I can't POST with my ID to create the thing even though it is a user-supplied ID. Consider combining create and update into one operation using resource version to differentiate.

so we make a distinction between `group_id` and `group_name`. the group id is an uuid generated by us, the group_name is a string supplied by the customer.

when creating a user group, the customer can pass in the group name as part of the UserGroupSpec

Reply:
Works for me and seems to be inline w/ what we do for users

Reply:
How does the create/update of user work? Can the user specify the ID? Should the ID be on the spec?

Reply:
Confirmed off-PR that this works just like user creation/update